extra/opensslpp/include_private/opensslpp/rsa_padding_conversions.hpp

```diff
@@ -0,0 +1,45 @@
+/* Copyright (c) 2022 Percona LLC and/or its affiliates. All rights reserved.
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU General Public License
+   as published by the Free Software Foundation; version 2 of
+   the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA */
+
+#ifndef OPENSSLPP_RSA_PADDING_CONVERSIONS_HPP
+#define OPENSSLPP_RSA_PADDING_CONVERSIONS_HPP
+
+#include <cassert>
+
+#include <openssl/rsa.h>
+
+#include <opensslpp/rsa_padding.hpp>
+
+namespace opensslpp {
+
+inline int rsa_padding_to_native_padding(rsa_padding padding) noexcept {
```

Comment:
Again, this is a drawback of the `public headers never include openssl headers` rule.
In the public header file we cannot write
enum class rsa_padding {
  no = RSA_NO_PADDING,
  pkcs1 = RSA_PKCS1_PADDING
};
because RSA_NO_PADDING and RSA_PKCS1_PADDING are not available there.